cni/pkg/nodeagent/net.go

```diff
@@ -287,7 +293,8 @@ func (s *NetServer) syncHostIPSets(ambientPods []*corev1.Pod) error {
 		} else {
 			err := addPodToHostNSIpset(pod, podIPs, &s.hostsideProbeIPSet)
 			if err != nil {
-				return err
+				// nolint: lll
+				log.Errorf("pod %s has IPs %v, but previously inserted one of those IPs for a different pod, this indicates an IPAM problem, pod will be skipped and will fail healthchecks", pod.Name, podIPs)
 			}
 			addedIPSnapshot = append(addedIPSnapshot, podIPs...)
```

Comment:
do we want to `return` early? otherwise we are adding it here but its not really added.. Or rather, probably already added

Reply:
No, if we return early we will skip any other pods in the snapshot that may not be dupes.

Reply:
Do we want to add to addedIPSnapshot though?

Reply:
It's already there so it doesn't matter really for what we're using it for, I'll tidy tho.